codegen/src/test/resources/software/amazon/awssdk/codegen/poet/batchmanager/test-batchmanager-async-interface.java

```diff
@@ -0,0 +1,114 @@
+package software.amazon.awssdk.services.batchmanagertest.batchmanager;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.services.batchmanagertest.BatchManagerTestAsyncClient;
+import software.amazon.awssdk.services.batchmanagertest.batchmanager.internal.DefaultBatchManagerTestAsyncBatchManager;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestResponse;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+/**
+ * Batch manager class that implements automatic batching features for a BatchManagerTest async client. This can be
+ * created using the static {@link #builder()} method.
+ * <p>
+ * The batch manager's automatic batching features allows for request batching using client-side buffering. This means
+ * that calls made from the client are first buffered and then sent as a batch request to the service. Client side
+ * buffering allows buffering a number of requests up to a service or user defined limit before being sent as a batch
+ * request. Outgoing calls are also periodically flushed after a defined period of time if batch requests do not reach
+ * the defined batch size limit.
+ */
+@Generated("software.amazon.awssdk:codegen")
+@SdkPublicApi
+public interface BatchManagerTestAsyncBatchManager extends SdkAutoCloseable {
+    /**
+     * Buffers outgoing {@link SendRequestRequest}s on the client and sends them as a
+     * {@link software.amazon.awssdk.services.batchmanagertest.model.SendRequestBatchRequest} to BatchManagerTest.
```

Comment:
Is there any reason this is fully qualified?

Reply:
So I tried fixing this earlier but it seems to be something to do with how poet references types using $T. It seems to be that poet only imports types when they are used in the code. Since `SendRequestBatchRequest` is only used in the javadocs, it doesn't add an import statement for the type. And since you can't manually add non-static imports, poet just automatically fills it in with the fully qualified name. On the other hand, `SendRequestRequest` is just filled in with the simple name since it is used elsewhere in the code and so is already imported. 

I also tried just emitting a string literal for `SendRequestBatchRequest` but it won't link to the actual class.

Reply:
Got it. Thanks for the clarification